### ISSUE TITLE
Loading blend files with lights in second runs moves them in-game.

### ISSUE THREAD

**Issue by Clockwork-Muse** [Moguri/blend2bam#44]
Sorta.

Taking a blend file and loading it via `loader.loadModel("reference/to/the/file.blend")`, the _second_ time the program is run the light is ...missing.  Or something.  It's no longer having the same effect on the render result, but the listed position from printing the tree hasn't changed. 

This only seems to have an effect on "live" `.blend` files - manually packing the `.bam` file for distribution works fine.  Saving the file resets the issue.  So whatever is going is almost certainly related to the conversion cache, but it's not clear why.

First run: 
![image](https://user-images.githubusercontent.com/5569508/100807948-14d71000-33e8-11eb-8f60-136a28f887c6.png)

```
ModelRoot Scene S:(LightAttrib)
  PandaNode Cube
    GeomNode Cube (1 geoms: S:(CullFaceAttrib MaterialAttrib TextureAttrib))
  PandaNode Light T:m(pos 4.07625 1.00545 5.90386 hpr 104.532 37.1946 3.97274)
    PointLight Light ( PerspectiveLens PerspectiveLens PerspectiveLens PerspectiveLens PerspectiveLens PerspectiveLens ):
      color 1000 1000 1000 1000
      attenuation 1 0 1
  PandaNode Camera T:m(pos 7.35889 -6.92579 4.95831 hpr 46.6919 63.5593 3.05786e-06)
    Camera Camera ( PerspectiveLens )
      PerspectiveLens fov = 40.7026 22.8952
```
Second run:
![image](https://user-images.githubusercontent.com/5569508/100807973-23bdc280-33e8-11eb-8c3b-558661d95d93.png)
```
ModelRoot Scene S:(LightAttrib)
  PandaNode Cube
    GeomNode Cube (1 geoms: S:(CullFaceAttrib MaterialAttrib TextureAttrib))
  PandaNode Light T:m(pos 4.07625 1.00545 5.90386 hpr 104.532 37.1946 3.97274)
    PointLight Light ( PerspectiveLens PerspectiveLens PerspectiveLens PerspectiveLens PerspectiveLens PerspectiveLens ):
      color 1000 1000 1000 1000
      attenuation 1 0 1
  PandaNode Camera T:m(pos 7.35889 -6.92579 4.95831 hpr 46.6919 63.5593 3.05786e-06)
    Camera Camera ( PerspectiveLens )
      PerspectiveLens fov = 40.7026 22.8952
```

<hr />

Changing the light type from point to directional (Sun) makes it imply that the light is rotated around somewhere:
![image](https://user-images.githubusercontent.com/5569508/100808867-cd518380-33e9-11eb-9d03-378a78c2b07c.png)

```
ModelRoot Scene S:(LightAttrib)
  PandaNode Cube
    GeomNode Cube (1 geoms: S:(CullFaceAttrib MaterialAttrib TextureAttrib))
  PandaNode Light T:m(pos 4.07625 1.00545 5.90386 hpr 104.532 37.1946 3.97274)
    DirectionalLight Light ( OrthographicLens ):
      color 1000 1000 1000 1000
      direction 0 1 0
  PandaNode Camera T:m(pos 7.35889 -6.92579 4.95831 hpr 46.6919 63.5593 3.05786e-06)
    Camera Camera ( PerspectiveLens )
      PerspectiveLens fov = 40.7026 22.8952
```

![image](https://user-images.githubusercontent.com/5569508/100808883-d93d4580-33e9-11eb-98fb-7fa4c3e7a231.png)

```
ModelRoot Scene S:(LightAttrib)
  PandaNode Cube
    GeomNode Cube (1 geoms: S:(CullFaceAttrib MaterialAttrib TextureAttrib))
  PandaNode Light T:m(pos 4.07625 1.00545 5.90386 hpr 104.532 37.1946 3.97274)
    DirectionalLight Light ( OrthographicLens ):
      color 1000 1000 1000 1000
      direction 0 1 0
  PandaNode Camera T:m(pos 7.35889 -6.92579 4.95831 hpr 46.6919 63.5593 3.05786e-06)
    Camera Camera ( PerspectiveLens )
      PerspectiveLens fov = 40.7026 22.8952
```

<hr />

Trivial program file:

```python
"""
Quick test module
"""
from panda3d.core import loadPrcFileData
from direct.showbase.ShowBase import ShowBase

class MyApp(ShowBase):
    """
    Quick test class
    """

    def __init__(self):
        ShowBase.__init__(self)

        # Load the environment model.
        self.scene = self.loader.loadModel("untitled.blend")
        # Reparent the model to render.
        self.scene.reparentTo(self.render)
        self.scene.setPos(0, 0, 0)
        # Modify the default camera
        self.disableMouse()
        #self.useDrive()
        self.camera.setPos(7.3589, -6.9258, 4.9583)
        self.camera.lookAt(0, 0, 0)

        self.scene.ls()

def generate_synth():
    """
    Quick test method
    """

    loadPrcFileData("",
    """
        textures-power-2 none
        audio-library-name null # Prevent ALSA errors
        show-frame-rate-meter 0
        sync-video 0
    """ )

    base = MyApp()
    base.run()

generate_synth()
```

**Comment by rdb** [comment]
If you disable the model cache, by setting `model-cache-dir` (without arguments) in Config.prc, does the problem go away?

**Comment by Clockwork-Muse** [comment]
Yes

**Comment by Clockwork-Muse** [comment]
I suppose I should include the following:

Blender: 2.90/2.91
Panda: 1.10.7
blend2bam 0.17

I'm running this inside a vscode .devcontainer, but that shouldn't be making a difference.

**Comment by rdb** [comment]
It may be an issue with Panda's model cache then, could you add a `self.scene.writeBamFile('scene.bam')` call and run it after loading the file for the first time (or with model cache disabled)?

Then, try loading the resulting bam file and see if the problem is reproduced.  If so, please share the bam file.

**Comment by Clockwork-Muse** [comment]
Issue is present in saved bam file, regardless of whether the model cache is disabled.

There's no difference in the information printed by `self.scene.ls()`
[scene.bam.zip](https://github.com/Moguri/blend2bam/files/5638407/scene.bam.zip)

**Comment by rdb** [comment]
Thanks, and can you share the .blend file as well for comparison?

**Comment by Clockwork-Muse** [comment]
[scene.blend.zip](https://github.com/Moguri/blend2bam/files/5638862/scene.blend.zip)
 (Although it should be the default blend file)

**Comment by rdb** [comment]
I think this is a bug in Panda3D, and I will look into fixing it.  You can use this as a workaround:
```python
scene.clearLight()

for light in scene.findAllMatches("**/+Light"):
    scene.setLight(light)
```

**Comment by rdb** [comment]
I don't think this issue can be fixed in Panda3D 1.10.8 (but will need to be in 1.11), but a workaround can be implemented in blend2bam, see #45.

**Comment by Clockwork-Muse** [comment]
Thanks!
